lib/compat/wordpress-6.7/formatting.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Extensions to the WordPress Formatting API.
+ * src/wp-includes/formatting.php
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Given a numeric value, returns a rounded a valid CSS <number> as a string.
+ * Negative zero values, e.g., `-0.0`, will return "0".
+ * See https://developer.mozilla.org/en-US/docs/Web/CSS/number
+ * Normalizes LM_NUMERIC locales with non-dot decimal point, e.g., "1,234.56" to "1234.56".
+ * The above is relevant to PHP < 8, whose float to string casting is locale-dependent.
+ * See https://php.watch/versions/8.0/float-to-string-locale-independent
+ *
+ * Usage:
+ * Call just before constructing the final output of your CSS rules, and after any number-based calculation.
+ *
+ * $padding_rounded = gutenberg_round_css_value( 20.3566 + 5, array( 'decimal_places' => 1 ) );
+ * echo "padding: {$padding_rounded}px;" // "padding: 25.4px;"
+ *
+ * @param int|string|float $value A CSS <number> data type.
```

Comment:
Good points, I agree it would reduce complexity. 

I think I accepted strings mainly out of convenience, so folks consuming theme.json values wouldn't have to coerce before using this function.

But I added a note about the [CSS <number> ](https://developer.mozilla.org/en-US/docs/Web/CSS/number), so on a bizarre subconscious level, I was halfway there. 😄